test/lastTraining/lastTraining.test.ts

```diff
@@ -0,0 +1,100 @@
+import { TestDatabase } from '../../src/helpers/testhelpers';
+import { describe, test, beforeEach, afterEach } from 'mocha';
+import chai, { expect } from 'chai';
+import supertest from 'supertest';
+import { setupServer } from '../../src/server';
+import { formatISO, subDays } from '../../src/helpers/dates';
+import { TrainingSession } from '../../src/schemas/training.session';
+import deepEqualInAnyOrder from 'deep-equal-in-any-order';
+
+chai.use(deepEqualInAnyOrder);
+
+describe('LastTraining Endpoint Tests', () => {
+  const testdb = new TestDatabase();
+  const testserver = supertest(setupServer(true));
+
+  beforeEach(async () => {
+    try {
+      await testdb.initialize();
+    } catch (e) {
+      console.log(e);
+    }
+  });
+
+  test('Get Method 200 w/ no day trained', async () => {
+    const res = await testserver.get('/lastTraining?days=7');
+    const now = Date.now();
+    const notTrainedLast7Days = [];
+    for (let index = 0; index < 7; index++) {
+      notTrainedLast7Days.push({
+        date: `${formatISO(subDays(now, index)).split('T')[0]}`,
+        trained: false,
+      });
+    }
+    expect(res.status).to.equal(200);
+    expect(res.body.data).to.deep.equalInAnyOrder(notTrainedLast7Days);
+  });
+
+  test('Get Method 400 bc of negative value', async () => {
+    const res = await testserver.get('/lastTraining?days=-7');
+    expect(res.status).to.equal(400);
+  });
+
+  test('Get Method 400 bc of missing value', async () => {
+    const res = await testserver.get('/lastTraining');
+    expect(res.status).to.equal(400);
+  });
+
+  test('Get Method 200 w/ one day trained', async () => {
+    const now = Date.now();
+
+    await TrainingSession.create({
+      _id: '5099803df3f4948bd2f98548',
+      userId: '5099803df3f494add2f9dja5',
+      trainingDayId: '5099803df3f4948bd2f9dja5',
+      date: `${formatISO(now)}`,
+      executions: [
+        {
+          exercise: {
+            _id: '5099803df3f4948bd2f98391',
+            name: 'Bench Press',
+            instruction: 'Push the bar.',
+            gifUrl: 'http://d205bpvrqc9yn1.cloudfront.net/0030.gif',
+            muscle: 'breast',
+            equipment: 'barbell',
+          },
+          notes: ['string'],
+          sets: [
+            {
+              executionType: 'warmup',
+              weight: 0,
+              reps: 0,
```

Comment:
Du willst hier eigentlich nicht die ganzen Unterobjekte ganz reinschreiben, weil ihr ja eigene Collections für Excercises habt, sondern nur die id angeben, das geht aber aktuell noch nicht, weil ihr keine Referenzen auf andere Models in den Schemas benutzt habt sondern da auch einfach die ganzen Models nochmal reingeschrieben habt, da habe ich gestern auch schonmal mit @juengeja drüber gesprochen. (Referenzen auf andere Models sind im entferntesten Sinne ähnlich wie Foreign Keys in einer relationalen DB) [https://stackoverflow.com/questions/18001478/referencing-another-schema-in-mongoose](https://stackoverflow.com/questions/18001478/referencing-another-schema-in-mongoose)

Reply:
würde jetzt ungern in die Types von Jannis pfuschen, während er damit arbeitet

Reply:
Fair point, kannst ja warten bis er seins gemerged hat und es dann bei dir reinmergen oder dich sonst irgendwie mit ihm absprechen